### DISCUSSION TITLE
How to deal with a persisted state shared between different systems

### DISCUSSION THREAD

**Discussion by cryptosmithio** [statelyai/xstate#3620] [Q&A]
I have an object that I am persisting to a DB.  Assume the object is observable.  In fact, everything in the DB is easily observable (RXDB).  Call it Object A.

When a user comes to my app, I use a machine to load the object and assign it as the context to my machine.  Call it Machine A.

When I assign Object A as context to Machine A, I use a series of conditions to determine what state the machine should be after the loading. Something like this:

```js
'link loaded': {
        always: [
          {
            cond: 'objectUnclaimed',
            target: 'unclaimed'
          },
          {
            cond: 'objectClaimed',
            target: 'claimed'
          },
          {
            cond: 'objectCancelled',
            target: 'cancelled'
          },
          {
            cond: 'objectFinalized',
            target: 'finalized'
          }
        ]
      }
```

```js
      guards: {
        objectUnclaimed: (context) => context.data.status === Status..unclaimed,
        objectClaimed: (context) => context.data.status === Status.claimed,
        objectCancelled: (context) => context.data.status === Status.cancelled,
        objectFinalized: (context) => context.data.status === Status.finalized
      },
```

I am not sure if that's the correct approach, determining my state from the status of the object.  However, the reason I am doing it is there are going to be services running on the back-end of our architecture which will need to change the state of Machine A.  Perhaps updating a database row for example.

What's the best approach?  I need to basically hydrate Machine A with the context of Object A.  Then either the user could send events directly to Machine A to change, but it is possible that a process on another machine needs to change the state.  For example, maybe somebody needs to "cancel" the object from another app.  

I can easily fire these events using RXDB and Observable, but I'm not sure where I should keep state.  I know I shouldn't manipulate an object being used as a context to a Machine directly. 

In short:
1. What's the best way to load context from a data store and apply it to a machine?
2. How should other processes communicate events to a machine via an Observable?

Thanks!  New to Xstate but exciting to build our system on it.

**Comment by davidkpiano** [comment]
Hey, thanks for your question! Do you have a more full code example we can work from?
